addons/stock/migrations/8.0.1.1/post-migration.py

```diff
@@ -530,12 +531,19 @@ def get_location_by_usage(usage):
 def migrate_stock_warehouses(cr, registry):
     """Migrate all the warehouses"""
     warehouse_obj = registry['stock.warehouse']
+    # Add a code to all warehouses that have no code
+    openupgrade.logged_query(
+        cr, """
+        UPDATE stock_warehouse SET code= 'WH' || to_char(id, 'FM999MI')
```

Comment:
The *|| to_char...* bit effectively appends the warehouse id to the code, so it is different per warehouse.